src/executors/ParallelExecutor.js

```diff
@@ -1,19 +1,52 @@
 import ExecutorAbstract from '../ExecutorAbstract';
+import ExecutorFactory from '../ExecutorFactory';
+import {wrapIfNotInstance} from '../util/type';
 
 class ParallelExecutor extends ExecutorAbstract {
+  //-----------------------------------
+  // Constructor
+  //-----------------------------------
+  /**
+   * Constructor for a new ParallelExecutor
+   *
+   * @param thread The parent thread that owns this executor.
+   * @param executorArray The executors to run in parallel
+   */
   constructor(thread, executorArray) {
     super(thread);
 
     this.executorArray = executorArray;
+    this.executorFactoryArray = executorArray.map((e) => wrapIfNotInstance(e, ExecutorFactory));
+    this.awaitingExecutors = executorArray.length;
   }
 
+  //-----------------------------------
+  // Methods
+  //-----------------------------------
+  /**
+   * Internal execution method called by Thread only.
+   *
+   * @param doneHandler The handler to call when done() is called.
+   * @param failHandler The handler to call when fail() is called;
+   * @private
+   */
   _execute(doneHandler, failHandler) {
     super._execute(doneHandler, failHandler);
-    // throw.Error('Parallels have not been implemented yet');
+    this.executorFactoryArray.forEach((executorFactory) => {
+      executorFactory.build(this.thread)._execute(this._parallelDone.bind(this), this._parallelFail.bind(this));
+    });
   }
 
-  toString() {
-    return this.id + ': ' + this.condition.toString().substring(0,64);
+  _parallelDone() {
+    this.awaitingExecutors -= 1;
+
+    if (this.awaitingExecutors === 0) {
+      this.done();
+    }
+  }
+
+  _parallelFail(error) {
+    this.fail(error);
```

Comment:
This look great! Only thing I might add is a option object that can be passed in as the final item in the parallels array. One of the options on that object can be 'killThreadOnFail', and we use that to determine whether to pass the kill prop to this.fail(error) here.